cpp/dolfinx/common/IndexMap.h

```diff
@@ -145,9 +147,12 @@ class IndexMap
 
   /// @todo Reconsider name
   /// Local (owned) indices shared with neighbor processes, i.e. are
-  /// ghosts on other processes
+  /// ghosts on other processes, grouped by sharing (neighbor)
+  /// process(destination ranks in forward communicator and source ranks in the
+  /// reverse communicator)
   /// @return List of indices that are ghosted on other processes
-  const std::vector<std::int32_t>& shared_indices() const noexcept;
+  const std::unique_ptr<graph::AdjacencyList<std::int32_t>>&
```

Comment:
If `_shared_indices` is always created in the constructor it would be better to return a reference here.